tile-join.cpp

```diff
@@ -642,12 +643,24 @@ struct tileset_reader {
 
 		long long scale = (1LL << zoom) / (1LL << maxzoom_so_far);
 
-		for (auto const &xy : tiles_at_maxzoom_so_far) {
-			for (long long xx = 0; xx < scale; xx++) {
-				for (long long yy = 0; yy < scale; yy++) {
-					overzoomed_tiles.push_back(std::pair<unsigned, unsigned>(xy.first * scale + xx, xy.second * scale + yy));
+		// If this is the first overzoomed level, we don't know yet
+		// which tiles will be useful, so spell out all 4 child tiles
+		// from each parent tile.
+		//
+		// If it is further overzoomed than that, we have a list of
+		// which child tiles will have features in them, so use that.
+
+		if (zoom == maxzoom_so_far + 1) {
+			for (auto const &xy : tiles_at_maxzoom_so_far) {
+				for (long long xx = 0; xx < scale; xx++) {
+					for (long long yy = 0; yy < scale; yy++) {
+						overzoomed_tiles.push_back(std::pair<unsigned, unsigned>(xy.first * scale + xx, xy.second * scale + yy));
+					}
 				}
 			}
+		} else {
+			overzoomed_tiles = next_overzoomed_tiles;
```

Comment:
Let's use `std::move` here since the size of these vectors _is_ the memory problem we're trying to fix, so it's better if we can avoid temporarily duplicating them in memory.

Reply:
OK, fixed